### ISSUE TITLE
[docs] Missing documentation regarding Expo Background Fetch

### ISSUE THREAD

**Issue by BrodaNoel** [expo/expo#32220]
### Summary

I was checking that in the TaskManager documentation is mentioning that BackgroundFetch needs to add a key in the infoPlisst, which is not defined on the BackgorunfFetch documentation.

So, in https://docs.expo.dev/versions/latest/sdk/task-manager/ you can see this:
<img width="828" alt="Screen Shot 2024-10-22 at 01 45 10" src="https://github.com/user-attachments/assets/fc0de637-6667-40da-9998-fc2fc75d6ec4">

But then, in here https://docs.expo.dev/versions/latest/sdk/background-fetch/ there is no information about that `"fetch"` value (and this may be the reason why in the last years this code was not working at all... Thanks, ChatGPT).

### Link to the related docs page

https://docs.expo.dev/versions/latest/sdk/background-fetch/

**Comment by mrakesh0608** [comment]
@BrodaNoel 

Check here : https://developer.apple.com/documentation/xcode/configuring-background-execution-modes#Specify-the-background-modes-your-app-requires

**Comment by chrfalch** [comment]
@BrodaNoel Thanks for reporting this, the docs are absolutely a bit misleading here.

When you install `expo-background-fetch` it contains a config plugin that inserts the correct `UIBackgroundMode` setting into the iOS project's `Info.plist` file so there shouldn't be necessary to do this manually. 

I just tested this by creating a new Expo app using SDK 51 where I installed `expo-background-fetch`. After running prebuild (Expo Go is already set up with the correct UIBackgroundModes) the following is added to my `Info.plist`:

```
 <key>UIBackgroundModes</key>
 <array>
   <string>fetch</string>
 </array>
```

Remember to run prebuild after installing the `expo-background-fetch` package: `npx expo prebuild -p ios`

This should be enough to get the correct `UIBackgroundMode` set in your project.

I'll go through the docs and see if I can clarify this a bit.
